### PR TITLE
Space Traveler's Hub

### PR DESCRIPTION
**Pull request by Yerimah**
In this development branch we;

- Fetched the data for rockets, missions, and dragons from the SpaceX live data API
- Rendered the data to the UI
- Unit tested Rockets, Missions, Dragon, and My profile components
- Implemented functionalities for activating and canceling rockets and dragon reservations, mission selection
- Displayed the reservations made and mission selected on the MY PROFILE page
- Fixed linter errors and resolved merge conflicts

### CODE REVIEW

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!

### HIGHLIGHTS!

- No linters errors.💯
- Descriptive Readme.md. 💯


There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    "react-test-renderer": "^17.0.2",
+    "redux": "^4.1.2",
+    "redux-logger": "^3.0.6",
+    "redux-thunk": "^2.4.1",
+    "web-vitals": "^2.1.4"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
```
[the remark this answers is not in the document]
Comment (Yerimah) [reply package.json:25]: I have written the tests and added the test files

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @Yerimah, @BertrandConxy, @zieeco 👋🏻, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️

- [ ] Some tests do not pass as seen below 👇🏻! Kindly try to refactor them to make them pass

image


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Components/Missions/Missions.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                  }
+                >
+                  {reserved ? 'ACTIVE MEMBER' : 'NOT A MEMBER'}
+                </p>
+              </td>
+              <td key={mission_id} className={styles.badge}>
+                <button
+                  className={reserved ? styles.leave : styles.join}
+                  data-target={mission_id}
+                  key={mission_id}
```
Comment (MrBrN197) [inline src/Components/Missions/Missions.js:56]: - [ ] [OPTIONAL] avoid using eslint-disable comments to get around ESlint errors. all you would have to do in this case is rename variables 

```suggestion
            mission_id: missionId,
            description,
            reserved,
          }) => (
            <tr key={missionId}>
              <td className={styles.missionName}>{missionName}</td>
              <td>{description}</td>
              <td className={styles.badge}>
                <p
                  className={
                    reserved ? styles.missionReserved : styles.notReserved
                  }
                >
                  {reserved ? 'ACTIVE MEMBER' : 'NOT A MEMBER'}
                </p>
              </td>
              <td key={missionId} className={styles.badge}>
                <button
                  className={reserved ? styles.leave : styles.join}
                  data-target={missionId}
                  key={missionId}
```

File: src/Components/MyProfile/MyProfile.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import styles from './myProfile.module.css';
+
+const MyProfile = () => {
+  const allDragons = useSelector((state) => state.dragonReducer);
+  const missionsState = useSelector((state) => state.missionsReducer.missions);
+  const joinedMissions = missionsState.filter(
+    (missions) => missions.reserved === true,
+  );
+
+  const reservedDragons = allDragons.filter((dragons) => !dragons.reserved !== true);
```
Comment (MrBrN197) [inline src/Components/MyProfile/MyProfile.js:12]: - [ ] [OPTIONAL] you can add a message that states there are no available reserved dragons/missions when they are no reservered dragons/missions. that way it is clear to the user that this is the case.

**Review by MrBrN197 — APPROVED** [review]
## Hi, @zieeco & @Yerimah  👋

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Yerimah** [comment]
Thank you

File: src/Components/MyProfile/MyProfile.js (hunk 3 above)
Comment (zieeco) [reply src/Components/MyProfile/MyProfile.js:12]: Thanks, man!  we'd put that in mind for other cases like that

**Pull request merged**